### DISCUSSION TITLE
is there somebody to use Debezium to create an connector to reading the database change record?

### DISCUSSION THREAD

**Discussion by linxd** [nextgenhealthcare/connect#5266] [Feature/Enhancement Ideas]
The Debezium  is powerful, want to use it to implement a source connector,  can share some idea?

**Comment by jonbartels** [comment]
The usual pattern for any custom connector is:

* Get the Java library that allows that connector to be called. Does Debezium have a Java SDK or library?
* Load the Java library as a Resource in Mirth Connect
* Write a Javascript poller connector that calls the Java library, it should produce a String that represents the message you want to pass through Mirth Connect

Another pattern is if Debezium can output HTTP or TCP messages then MC can just be an ordinary receiver and listen for those messages.

**Reply by pacmano1** [reply]
Also a cross post to the forums: https://forums.mirthproject.io/forum/mirth-connect/support/181636-use-debezium-to-implement-a-source-connector#post181642

**Reply by ab-20-m** [reply]
A quick google-ing says Debezium monitors DB changes and writes change events to Kafka topics. I think someone asked about a Kafka-JMS connector a while back. 

So a quick napkin design would be something like:

1. JMS reader channel 
2. With directory resource to load the Confluent JMS-Kafka library and dependencies
3. Configured to point at the Kafka instance backing the Debezium instance and the topic you want change events from

Or as @jonbartels said create a Javascript poller and connect with the Debezium SDK and get events that way.

**Reply by linxd** [reply]
thanks.  How about add an extension of  new type source connector of debezium connector?

**Reply by jonbartels** [reply]
https://github.com/kpalang/mirth-plugin-guide is an excellent guide on how to write MC plugins.

**Reply by pacmano1** [reply]
If you are asking for Nextgen to write a plugin, you should create a feature reqeust under "issues" here on  Github.    In my opinion though, I would not expect Nextgen to create a connector for Debezium.
